### PR TITLE
Fixes slime attacks not getting logged

### LINKED ISSUE #65020: Slime attacking objects isn't logged
Reporting client version: 514.1578

## Round ID:
[178650](https://scrubby.melonmesa.com/round/178650)

## Reproduction:

A gondola, a slime, and an oxygen canister walk into a thunderdome.

Be the slime.  Beat up the gondola and oxygen tank.  Observe that the gondola hits get logged and the oxygen tank hits don't.

( I initially confused myself because e.g. spiders log ok, apparently simple mobs have various special little branches that aren't logged, slimes being an example of this, and there may be others.  But this is the one I know about after failing to forensics something via logging in an active round )

### PR DESCRIPTION
**Pull request by ghost**
## About The Pull Request

Fixes https://github.com/tgstation/tgstation/issues/65020

Now when slimes attack objectis it will be logged. So, if a player-controlled slime sucessfully attacks an object, it will be logged as "attacked", and if it fails to attack something due to being a non-adult slime or for other reason it will be logged as "attempted to attack"(can be usefull to predict things like non-antag slimes destroying the SM or something like this).

## Why It's Good For The Game

Admins can more properly deal with griefing slimes

## Changelog

:cl: SuperSlayer
fix: fixes slimes attacking objects not getting logged
/:cl:

### CODE REVIEW

File: code/game/objects/obj_defense.dm [hunk 1]
```diff
@@ -95,8 +95,14 @@
 
 /obj/attack_slime(mob/living/simple_animal/slime/user, list/modifiers)
 	if(!user.is_adult)
+		if(user.client)
+			log_combat(user, src, "attempted to attack")
```
Comment (ShizCalev) [inline code/game/objects/obj_defense.dm:99]: why does this need to be logged

File: code/game/objects/obj_defense.dm [hunk 2]
```diff
@@ -95,8 +95,14 @@
...
 		return
-	attack_generic(user, rand(10, 15), BRUTE, MELEE, 1)
+	if(user.client)
```
Comment (ShizCalev) [inline code/game/objects/obj_defense.dm:101]: NPC slimes use attack_slime() on objects that bump them.

**Review by ShizCalev — CHANGES_REQUESTED** [review]

File: code/game/objects/obj_defense.dm (hunk 1 above)
Comment (SuperSlayer0) [reply code/game/objects/obj_defense.dm]: To predict shit happening, but i can remove it

File: code/game/objects/obj_defense.dm (hunk 2 above)
Comment (SuperSlayer0) [reply code/game/objects/obj_defense.dm:99]: Soooo, what should i do then?

Comment (ShizCalev) [reply code/game/objects/obj_defense.dm:99]: Honestly I don't see a need to be checking for a client at all, just log the attack normally.

Comment (SuperSlayer0) [reply code/game/objects/obj_defense.dm:99]: I think that there is no reason for logging NPC attacks, but i can make it log even without a client

**Review by ShizCalev — APPROVED** [review]

**Pull request merged**